src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsExamplesAstConsistencyTest.java

```diff
@@ -0,0 +1,516 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2025 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.internal;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.JavaParser;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FileContents;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Ensures xdocs Java examples for the same check differ only by comments.
+ *
+ * <p>This test validates that all Example1.java, Example2.java, etc. files within
+ * the same directory have identical AST structure .
+ * When differences are found, it provides readable AST dumps to help
+ * identify where the examples diverge.
+ *
+ */
+public class XdocsExamplesAstConsistencyTest {
+
+    private static final Path XDOCS_ROOT = Path.of(
+            "src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle"
+    );
+
+    private static final String COMMON_PATH =
+            "src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/";
+
+    // This list is temporarily suppressed.
+    // Until: https://github.com/checkstyle/checkstyle/issues/XXXX
+    private static final Set<String> SUPPRESSED_DIRECTORIES = Set.of(
```

Comment:
Please do us a favor to allow suppression by part of path.
There are Examples that will forever stay different from others.
Reason is that there are examples to show how how properties works, and there are examples to show some in demand use case that users always wants but it is not very obvious how to do.

So examples are also collection of ready to use configs.